core/modules/indexers/backlinks-index.js

```diff
@@ -0,0 +1,108 @@
+/*\
+title: $:/core/modules/indexers/backlinks-indexer.js
+type: application/javascript
+module-type: indexer
+
+Indexes the tiddlers' backlinks
+
+\*/
+(function(){
+
+/*jslint node: true, browser: true */
+/*global modules: false */
+"use strict";
+
+
+function BacklinksIndexer(wiki) {
+	this.wiki = wiki;
+}
+
+BacklinksIndexer.prototype.init = function() {
+	this.index = null;
+}
+
+BacklinksIndexer.prototype.rebuild = function() {
+	this.index = null;
+}
+
+BacklinksIndexer.prototype._getLinks = function(tiddler) {
+	var links = [],
+            parser =  this.wiki.parseText(tiddler.fields.type, tiddler.fields.text, {}),
+	    checkParseTree = function(parseTree) {
+			for(var t=0; t<parseTree.length; t++) {
+				var parseTreeNode = parseTree[t];
+				if(parseTreeNode.type === "link" && parseTreeNode.attributes.to && parseTreeNode.attributes.to.type === "string") {
+					var value = parseTreeNode.attributes.to.value;
+					if(links.indexOf(value) === -1) {
+						links.push(value);
+					}
+				}
+				if(parseTreeNode.children) {
+					checkParseTree(parseTreeNode.children);
+				}
+			}
+		};
+	if(parser) {
+		checkParseTree(parser.tree);
+	}
+	return links;
+}
+
+BacklinksIndexer.prototype.update = function(updateDescriptor) {
+	if(!this.index) {
+		return;
+	}
+
+	var newLinks = [],
+	    oldLinks = [];
+
+	if(updateDescriptor.old.exists) {
+		oldLinks = this._getLinks(updateDescriptor.old.tiddler);
```

Comment:
Oh, I see. If `.tiddlerDelete()` is activated, the  old.tiddler is already overwritten in the store and caches are destroyed :/ 

https://github.com/Jermolene/TiddlyWiki5/blob/b258afea0dc564b737409d88c72ca4180f967799/boot/boot.js#L1129-L1133

.. That's unfortunate. ... So may be the `updateDescriptor` should provide this info, to avoid parsing for a deleted tiddler, which is complete waste of time. 

@Jermolene what do you think?